frontend/src/features/shared/receipt/useReceiptForm.ts

```diff
@@ -0,0 +1,134 @@
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { patients } from '@/constants/mock-data';
+
+const receiptFormSchema = z.object({
+  waitingNumber: z.string().length(2, {
+    message: '필수 입력 항목입니다.',
+  }),
+  name: z.string().min(2, {
+    message: '필수 입력 항목입니다.',
+  }),
+  gender: z.string(),
+  birth: z.date({
+    required_error: '유효한 생년월일을 입력해 주세요.',
+  }),
```

Comment:
PR에 남겨주신 문제는 리셋은 동작하는데 DatePicker에 date=undefined로 전달되고 변수가 초기화되지 않아서 발생하는 문제 같아요. 우선 생년월일 형식이 'YYYY-MM-DD'로 저장되어서 `z.string().date()` 메서드로 변경하고 DatePicker 컴포넌트를 수정하기보다 date, onChange props 넘겨줄 때 형식 바꾸는게 좋을 것 같아요! (💭 첫 방문일도 YYYY-MM-DD로 포맷을 통일하는게 좋을까요,,?)

```suggestion
  birth: z.string().date(),
```